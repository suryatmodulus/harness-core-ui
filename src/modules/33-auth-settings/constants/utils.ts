/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum AuthenticationMechanisms {
  SAML = 'SAML',
  OAUTH = 'OAUTH',
  LDAP = 'LDAP',
  USER_PASSWORD = 'USER_PASSWORD'
}

export const getSamlEndpoint = (accountId: string): string => {
  let url = window.location.href.split('/#')[0]?.replace('/ng', '')

  if (window.apiUrl) {
    if (window.apiUrl.startsWith('/')) {
      url = `${url}${window.apiUrl}`
    } else {
      url = window.apiUrl
    }
  }

  return `${url}/api/users/saml-login?accountId=${accountId}`
}
