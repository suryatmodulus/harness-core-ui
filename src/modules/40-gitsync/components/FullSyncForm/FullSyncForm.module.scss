/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.modalContainer {
  padding: var(--spacing-xxlarge) !important;

  .modalBody {
    :global {
      .bp3-label {
        color: var(--grey-600) !important;
      }

      .bp3-input-group input {
        width: 400px;
      }

      .bp3-disabled input {
        background-color: var(--grey-100) !important;
      }
    }

    .formBody {
      min-height: 480px;

      .formRow {
        display: flex;
      }
    }

    .branchSection {
      .branchInput {
        width: 280px !important;
        margin-left: var(--spacing-huge);
      }

      :global {
        .bp3-form-group {
          margin-top: var(--spacing-xsmall);
          margin-bottom: 0;

          .bp3-form-content {
            font-size: var(--font-size-normal);
            color: var(--grey-600) !important;

            .bp3-control-indicator {
              margin-left: 45px !important;
            }
          }
        }
      }
    }
  }
}

.branchSelector {
  padding-top: var(--spacing-xsmall);
  padding-left: var(--spacing-small);
  padding-bottom: var(--spacing-xsmall);

  :global {
    input[type='text'] {
      width: 300px !important;
      background: var(--primary-2);
    }
  }
}

.gitBranchSelectorPopover {
  :global {
    width: 300px !important;

    .bp3-menu {
      padding: 0;
      li {
        height: 32px;
        cursor: pointer;
        padding: var(--spacing-xsmall) var(--spacing-small);

        &:hover {
          background-color: var(--primary-2) !important;
          color: var(--grey-900) !important;
        }
      }
    }

    .Select--active {
      background-color: var(--primary-2) !important;
      color: var(--grey-900) !important;
    }
  }
}

.prTitle {
  margin-left: 45px;
}

.tooltip {
  :global {
    .bp3-popover-arrow-fill {
      fill: var(--red-100) !important;
    }
    .bp3-popover-content {
      background-color: var(--red-100);
    }
  }
}
