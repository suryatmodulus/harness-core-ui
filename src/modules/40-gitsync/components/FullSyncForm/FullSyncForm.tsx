/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useRef, useState } from 'react'
import { Popover, Position } from '@blueprintjs/core'
import {
  Container,
  Text,
  Color,
  Layout,
  Formik,
  FormikForm,
  FormInput,
  Button,
  SelectOption,
  Radio,
  Icon,
  ModalErrorHandler,
  ModalErrorHandlerBinding,
  PageSpinner,
  FontVariation,
  useToaster
} from '@harness/uicore'
import * as Yup from 'yup'
import { debounce, defaultTo, isEmpty } from 'lodash-es'
import type { FormikContext } from 'formik'
import { useParams } from 'react-router-dom'
import {
  GitSyncConfig,
  GitSyncFolderConfigDTO,
  GitBranchDTO,
  getListOfBranchesWithStatusPromise,
  ResponseGitBranchListDTO,
  GitFullSyncConfigRequestDTO,
  createGitFullSyncConfigPromise,
  triggerFullSyncPromise
} from 'services/cd-ng'
import { useStrings } from 'framework/strings'
import { useGitSyncStore } from 'framework/GitRepoStore/GitSyncStoreContext'
import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
import css from './FullSyncForm.module.scss'

interface FullSyncFormProps {
  orgIdentifier: string
  projectIdentifier: string
  isNewUser: boolean
}

interface ModalConfigureProps {
  onClose?: () => void
  onSuccess?: () => void
}

const FullSyncForm: React.FC<ModalConfigureProps & FullSyncFormProps> = props => {
  const { isNewUser = true, orgIdentifier, projectIdentifier, onClose, onSuccess } = props
  const config = {} as GitFullSyncConfigRequestDTO
  const { accountId } = useParams<AccountPathProps>()
  const { showSuccess } = useToaster()
  const { getString } = useStrings()

  const { gitSyncRepos, loadingRepos } = useGitSyncStore()
  const [rootFolderSelectOptions, setRootFolderSelectOptions] = React.useState<SelectOption[]>([])
  const [repoSelectOptions, setRepoSelectOptions] = React.useState<SelectOption[]>([])
  const [isNewBranch, setIsNewBranch] = React.useState(false)

  const [branches, setBranches] = React.useState<SelectOption[]>()
  const [modalErrorHandler, setModalErrorHandler] = React.useState<ModalErrorHandlerBinding>()
  const formikRef = useRef<FormikContext<GitFullSyncConfigRequestDTO>>()
  const [createPR, setCreatePR] = useState<boolean>(false) //used for rendering PR title
  const [disableCreatePR, setDisableCreatePR] = useState<boolean>()
  const [disableBranchSelection, setDisableBranchSelection] = useState<boolean>(true)

  const defaultInitialFormData: GitFullSyncConfigRequestDTO = {
    baseBranch: config?.baseBranch,
    branch: '',
    createPullRequest: false,
    newBranch: false,
    prTitle: getString('gitsync.deafaultSyncTitle'),
    repoIdentifier: config?.repoIdentifier || '',
    rootFolder: config?.rootFolder || '',
    targetBranch: ''
  }

  const handleBranchTypeChange = (isNew: boolean, formik: FormikContext<GitFullSyncConfigRequestDTO>): void => {
    const defaultBranch = gitSyncRepos.find(
      (repo: GitSyncConfig) => repo.identifier === formikRef.current?.values.repoIdentifier
    )?.branch
    if (isNewBranch !== isNew) {
      setIsNewBranch(isNew)

      formik.setFieldValue('branch', `${defaultBranch}-patch`)
      formik.setFieldTouched('branch', false)
    }
    formik.setFieldValue('targetBranch', isNew ? defaultBranch || '' : '')
    formik.setFieldTouched('targetBranch', false)
    formik.setFieldValue('createPullRequest', false)
    formik.setFieldTouched('createPullRequest', false)
    formikRef.current?.setFieldValue('createPullRequest', false)
    setDisableCreatePR(false)
    setCreatePR(false)
    setDisableBranchSelection(true)
  }

  const getRootFolderSelectOptions = (folders: GitSyncFolderConfigDTO[] | undefined): SelectOption[] => {
    return folders?.length
      ? folders.map((folder: GitSyncFolderConfigDTO) => {
          return {
            label: folder.rootFolder || '',
            value: folder.rootFolder || ''
          }
        })
      : []
  }

  useEffect(() => {
    if (projectIdentifier && gitSyncRepos?.length) {
      defaultInitialFormData.repoIdentifier = defaultInitialFormData.repoIdentifier || gitSyncRepos[0].identifier || ''
      const selectedRepo = gitSyncRepos.find(
        (repo: GitSyncConfig) => repo.identifier === defaultInitialFormData.repoIdentifier
      )
      defaultInitialFormData.baseBranch = selectedRepo?.branch
      fetchBranches(defaultInitialFormData.repoIdentifier)
      setRepoSelectOptions(
        gitSyncRepos?.map((gitRepo: GitSyncConfig) => {
          return {
            label: gitRepo.name || '',
            value: gitRepo.identifier || ''
          }
        })
      )

      setRootFolderSelectOptions(getRootFolderSelectOptions(selectedRepo?.gitSyncFolderConfigDTOs))

      const defaultRootFolder = selectedRepo?.gitSyncFolderConfigDTOs?.find(
        (folder: GitSyncFolderConfigDTO) => folder.isDefault
      )
      defaultInitialFormData.rootFolder = defaultInitialFormData.rootFolder || defaultRootFolder?.rootFolder || ''
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [gitSyncRepos, projectIdentifier])

  const fetchBranches = (repoIdentifier: string, query?: string): void => {
    getListOfBranchesWithStatusPromise({
      queryParams: {
        accountIdentifier: accountId,
        orgIdentifier,
        projectIdentifier,
        yamlGitConfigIdentifier: repoIdentifier,
        page: 0,
        size: 10,
        searchTerm: query
      }
    })
      .then((response: ResponseGitBranchListDTO) => {
        const branchesInResponse = response?.data?.branches?.content
        /* Show error in case no branches exist on a git repo at all */
        /* A valid git repo should have atleast one branch in it(a.k.a default branch) */
        if (!query && isEmpty(branchesInResponse)) {
          setDisableCreatePR(true)
          setDisableBranchSelection(true)
          modalErrorHandler?.showDanger(getString('common.git.noBranchesFound'))
          return
        }
        const branchOptions = branchesInResponse?.map((branch: GitBranchDTO) => {
          return { label: branch?.branchName, value: branch?.branchName }
        }) as SelectOption[]

        if (!isNewBranch && isEmpty(branchOptions)) {
          setDisableCreatePR(true)
          setDisableBranchSelection(true)
        } else {
          setBranches(branchOptions)
          setDisableCreatePR(false)
          setDisableBranchSelection(false)
        }
      })
      .catch(e => {
        modalErrorHandler?.showDanger(e.data?.message || e.message)
      })
  }

  const saveAndTriggerFullSync = async (fullSyncData: GitFullSyncConfigRequestDTO): Promise<void> => {
    const queryParams = {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier
    }
    modalErrorHandler?.hide()
    try {
      const fullSyncConfigData = await createGitFullSyncConfigPromise({
        queryParams,
        body: { ...fullSyncData, newBranch: isNewBranch }
      })
      if (fullSyncConfigData.status !== 'SUCCESS') {
        throw fullSyncConfigData
      }
      const triggerFullSync = await triggerFullSyncPromise({ queryParams, body: {} as unknown as void })
      if (triggerFullSync.status !== 'SUCCESS') {
        throw triggerFullSync
      }
      showSuccess(getString('gitsync.syncSucessToaster'))
      onSuccess?.()
    } catch (err) {
      modalErrorHandler?.showDanger(defaultTo(err.data?.message, err.message))
    }
  }

  const debounceFetchBranches = debounce((repoIdentifier: string, query?: string): void => {
    try {
      fetchBranches(repoIdentifier, query)
    } catch (e) {
      modalErrorHandler?.showDanger(e.data?.message || e.message)
    }
  }, 1000)

  const CreatePR = React.useMemo(() => {
    return (
      <>
        <Layout.Horizontal flex={{ alignItems: 'baseline', justifyContent: 'flex-start' }} padding={{ top: 'small' }}>
          <FormInput.CheckBox
            disabled={disableCreatePR}
            name="createPullRequest"
            label={getString('common.git.startPRLabel')}
            onChange={e => {
              formikRef.current?.setFieldValue('createPullRequest', e.currentTarget.checked)
              formikRef.current?.setFieldTouched('targetBranch', false)
              setCreatePR(e.currentTarget.checked)
              if (e.currentTarget.checked) {
                fetchBranches(formikRef.current?.values.repoIdentifier || '')
              } else {
                setDisableBranchSelection(true)
              }
            }}
          />
          {disableCreatePR ? (
            <Popover
              position={Position.TOP}
              content={
                <Text padding="medium" color={Color.RED_400}>
                  {getString('common.git.onlyDefaultBranchFound')}
                </Text>
              }
              isOpen={disableCreatePR}
              popoverClassName={css.tooltip}
            >
              <Container margin={{ bottom: 'xlarge' }}></Container>
            </Popover>
          ) : null}
          <FormInput.Select
            name="targetBranch"
            items={branches || []}
            disabled={disableBranchSelection || disableCreatePR}
            data-id="create-pr-branch-select"
            onQueryChange={(query: string) =>
              debounceFetchBranches(formikRef.current?.values.repoIdentifier || '', query)
            }
            selectProps={{ usePortal: true, popoverClassName: css.gitBranchSelectorPopover }}
            className={css.branchSelector}
          />
        </Layout.Horizontal>
        {createPR ? (
          <FormInput.Text name="prTitle" className={css.prTitle} label={getString('gitsync.PRTitle')} />
        ) : null}
      </>
    )
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [
    disableCreatePR,
    disableBranchSelection,
    branches,
    isNewBranch,
    formikRef.current?.values,
    formikRef.current?.values.createPullRequest
  ])

  return (
    <Container height={'inherit'} className={css.modalContainer}>
      {loadingRepos ? (
        <PageSpinner />
      ) : (
        <>
          <Text font={{ variation: FontVariation.H3 }} padding={{ bottom: 'xlarge' }}>
            {getString('gitsync.fullSyncTitle')}
          </Text>
          <ModalErrorHandler bind={setModalErrorHandler} />
          <Container className={css.modalBody}>
            <Formik<GitFullSyncConfigRequestDTO>
              initialValues={defaultInitialFormData}
              formName="fullSyncConfigForm"
              validationSchema={Yup.object().shape({
                repoIdentifier: Yup.string().trim().required(getString('common.validation.repositoryName')),
                branch: Yup.string()
                  .trim()
                  .required(getString('validation.branchName'))
                  .when('createPullRequest', {
                    is: true,
                    then: Yup.string().notOneOf(
                      [Yup.ref('targetBranch')],
                      getString('common.git.validation.sameBranches')
                    )
                  }),
                targetBranch: Yup.string()
                  .trim()
                  .when('createPullRequest', {
                    is: true,
                    then: Yup.string().required(getString('common.git.validation.targetBranch'))
                  }),
                prTitle: Yup.string().trim().min(1).required(getString('common.git.validation.PRTitleRequired'))
              })}
              onSubmit={formData => {
                saveAndTriggerFullSync(formData)
              }}
            >
              {formik => {
                formikRef.current = formik
                return (
                  <FormikForm>
                    <Container className={css.formBody}>
                      <Text margin={{ bottom: 'small' }} font={{ size: 'medium' }} color={Color.GREY_700}>
                        {getString('common.gitSync.harnessFolderLabel')}
                      </Text>

                      <FormInput.Select
                        name="repoIdentifier"
                        label={getString('common.git.selectRepoLabel')}
                        items={repoSelectOptions}
                        disabled={isNewUser}
                      />
                      <FormInput.Select
                        name="rootFolder"
                        label={getString('common.gitSync.harnessFolderLabel')}
                        items={rootFolderSelectOptions}
                        disabled={isNewUser}
                      />

                      <Text font={{ variation: FontVariation.FORM_SUB_SECTION }} margin={{ top: 'large' }}>
                        {getString('gitsync.syncBranchTitle')}
                      </Text>
                      <Layout.Vertical spacing="medium" margin={{ bottom: 'medium' }}>
                        <Container
                          className={css.branchSection}
                          padding={{
                            top: 'small',
                            bottom: 'xSmall'
                          }}
                        >
                          <Layout.Horizontal flex={{ alignItems: 'baseline', justifyContent: 'flex-start' }}>
                            <Radio large onChange={() => handleBranchTypeChange(false, formik)} checked={!isNewBranch}>
                              <Icon name="git-branch-existing"></Icon>
                              <Text margin={{ left: 'small' }} inline>
                                {getString('gitsync.selectBranchTitle')}
                              </Text>
                            </Radio>
                            <FormInput.Select
                              name="branch"
                              items={branches || []}
                              disabled={isNewBranch}
                              onQueryChange={(query: string) =>
                                debounceFetchBranches(formik.values.repoIdentifier, query)
                              }
                              selectProps={{ usePortal: true, popoverClassName: css.gitBranchSelectorPopover }}
                              className={css.branchSelector}
                            />
                          </Layout.Horizontal>
                          {!isNewBranch && CreatePR}
                        </Container>
                        <Container
                          className={css.branchSection}
                          padding={{
                            bottom: isNewBranch ? 'xSmall' : 'small'
                          }}
                        >
                          <Radio
                            data-test="newBranchRadioBtn"
                            large
                            onChange={() => handleBranchTypeChange(true, formik)}
                            checked={isNewBranch}
                          >
                            <Icon name="git-new-branch" color={Color.GREY_700}></Icon>
                            <Text inline margin={{ left: 'small' }}>
                              {getString('gitsync.createBranchTitle')}
                            </Text>
                          </Radio>
                          {isNewBranch && (
                            <Container padding={{ top: 'small' }}>
                              <FormInput.Text
                                className={css.branchInput}
                                name="branch"
                                label={getString('common.git.branchName')}
                              />
                              {CreatePR}
                            </Container>
                          )}
                        </Container>
                      </Layout.Vertical>
                    </Container>

                    <Layout.Horizontal padding={{ top: 'medium' }} spacing="medium">
                      <Button type="submit" intent="primary" text={getString('save')} />
                      <Button text={getString('cancel')} margin={{ left: 'medium' }} onClick={onClose} />
                    </Layout.Horizontal>
                  </FormikForm>
                )
              }}
            </Formik>
          </Container>
        </>
      )}
    </Container>
  )
}

export default FullSyncForm
