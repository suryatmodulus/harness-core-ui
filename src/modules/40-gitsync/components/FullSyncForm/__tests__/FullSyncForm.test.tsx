/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, render, waitFor, queryByText } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { noop } from 'lodash-es'
import { GitSyncTestWrapper } from '@common/utils/gitSyncTestUtils'
import FullSyncForm from '../FullSyncForm'

const pathParams = { accountId: 'dummy', orgIdentifier: 'default', projectIdentifier: 'dummyProject' }

const fetchBranches = jest.fn(() => Promise.resolve([]))

jest.mock('services/cd-ng', () => ({
  createGitFullSyncConfigPromise: jest.fn().mockImplementation(() => noop()),
  triggerFullSyncPromise: jest.fn().mockImplementation(() => noop()),
  getListOfBranchesWithStatusPromise: jest.fn().mockImplementation(() => fetchBranches())
}))

describe('Test GitFullSyncForm', () => {
  test('Should render GitFullSyncForm', async () => {
    const { container } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
      >
        <FullSyncForm
          isNewUser={true}
          onClose={noop}
          onSuccess={noop}
          orgIdentifier={'default'}
          projectIdentifier={'dummyProject'}
        />
      </GitSyncTestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
  test('Should have new branch input hidden by default and it should appear when user selects it', async () => {
    const { container, getByText } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
      >
        <FullSyncForm
          isNewUser={true}
          onClose={noop}
          onSuccess={noop}
          orgIdentifier={'default'}
          projectIdentifier={'dummyProject'}
        />
      </GitSyncTestWrapper>
    )
    await waitFor(() => {
      expect(getByText('gitsync.createBranchTitle')).toBeDefined()
      const branchSelectorInput = container.querySelectorAll('input[name="branch"]') as unknown as HTMLInputElement[]
      expect(branchSelectorInput[0]).toBeTruthy()
      expect(branchSelectorInput[0].disabled).toBe(false)
      expect(queryByText(container, 'common.git.branchName')).toBeFalsy()
    })

    await act(async () => {
      const newBranchRadioBtn = document.querySelector('[data-test="newBranchRadioBtn"]') as HTMLInputElement
      fireEvent.click(newBranchRadioBtn!)
      expect(newBranchRadioBtn.value).toBe('on')
    })
  })

  test('Should have PR title hidden by default and it should appear when user selects create PR', async () => {
    const { container, getByText } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
      >
        <FullSyncForm
          isNewUser={true}
          onClose={noop}
          onSuccess={noop}
          orgIdentifier={'default'}
          projectIdentifier={'dummyProject'}
        />
      </GitSyncTestWrapper>
    )

    await waitFor(() => {
      expect(getByText('gitsync.createBranchTitle')).toBeDefined()
    })

    expect(container.querySelector('input[name="prTitle"]')).toBeFalsy()
  })
})
