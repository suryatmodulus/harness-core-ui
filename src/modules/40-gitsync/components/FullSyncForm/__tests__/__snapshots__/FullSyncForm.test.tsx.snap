// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GitFullSyncForm Should render GitFullSyncForm 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main modalContainer"
    style="height: inherit;"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--padding-bottom-xlarge"
    >
      gitsync.fullSyncTitle
    </p>
    <div
      class="StyledProps--font StyledProps--main modalBody"
    >
      <div
        class="OverlaySpinner--overlaySpinner"
      >
        <form
          class="FormikForm--main"
        >
          <div>
            <div
              class="StyledProps--font StyledProps--main formBody"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--margin-bottom-small StyledProps--font-size-medium StyledProps--color StyledProps--color-grey700"
              >
                common.gitSync.harnessFolderLabel
              </p>
              <div
                class="bp3-form-group bp3-disabled"
                data-id="repoIdentifier-0"
              >
                <label
                  class="bp3-label"
                  for="repoIdentifier"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="fullSyncConfigForm_repoIdentifier"
                  >
                    common.git.selectRepoLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-popover-wrapper Select--main bp3-fill"
                  >
                    <div
                      class="bp3-popover-target"
                    >
                      <div
                        class="bp3-input-group bp3-disabled"
                      >
                        <input
                          autocomplete="off"
                          class="bp3-input"
                          disabled=""
                          name="repoIdentifier"
                          placeholder="- Select -"
                          style="padding-right: 0px;"
                          type="text"
                          value="gitSyncRepoTest"
                        />
                        <span
                          class="bp3-input-action"
                        >
                          <span
                            class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                            icon="chevron-down"
                          >
                            <svg
                              data-icon="chevron-down"
                              height="14"
                              viewBox="0 0 16 16"
                              width="14"
                            >
                              <desc>
                                chevron-down
                              </desc>
                              <path
                                d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="bp3-form-group bp3-disabled"
                data-id="rootFolder-1"
              >
                <label
                  class="bp3-label"
                  for="rootFolder"
                >
                  <span
                    class="Tooltip--acenter"
                    data-tooltip-id="fullSyncConfigForm_rootFolder"
                  >
                    common.gitSync.harnessFolderLabel
                  </span>
                   
                  <span
                    class="bp3-text-muted"
                  />
                </label>
                <div
                  class="bp3-form-content"
                >
                  <div
                    class="bp3-popover-wrapper Select--main bp3-fill"
                  >
                    <div
                      class="bp3-popover-target"
                    >
                      <div
                        class="bp3-input-group bp3-disabled"
                      >
                        <input
                          autocomplete="off"
                          class="bp3-input"
                          disabled=""
                          name="rootFolder"
                          placeholder="- Select -"
                          style="padding-right: 0px;"
                          type="text"
                          value="/src1/.harness/"
                        />
                        <span
                          class="bp3-input-action"
                        >
                          <span
                            class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                            icon="chevron-down"
                          >
                            <svg
                              data-icon="chevron-down"
                              height="14"
                              viewBox="0 0 16 16"
                              width="14"
                            >
                              <desc>
                                chevron-down
                              </desc>
                              <path
                                d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-form-sub-section StyledProps--margin-top-large"
              >
                gitsync.syncBranchTitle
              </p>
              <div
                class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--margin-bottom-medium"
              >
                <div
                  class="StyledProps--font StyledProps--main branchSection StyledProps--padding-top-small"
                >
                  <div
                    class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline StyledProps--flex-justifyContent-flex-start"
                  >
                    <label
                      class="bp3-control bp3-radio bp3-large StyledProps--font Radio--radio StyledProps--main"
                    >
                      <input
                        checked=""
                        type="radio"
                      />
                      <span
                        class="bp3-control-indicator"
                      />
                      <span
                        class="bp3-icon StyledProps--main"
                        data-icon="git-branch-existing"
                      >
                        <svg
                          fill="none"
                          height="16"
                          viewBox="0 0 16 9"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M12.403 3.67H16v1.626h-3.597a4.477 4.477 0 01-8.807 0H0V3.67h3.596a4.477 4.477 0 018.807 0zm-7.26.813a2.857 2.857 0 105.714 0 2.857 2.857 0 00-5.714 0z"
                            fill="#383946"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="StyledProps--font StyledProps--main StyledProps--margin-left-small StyledProps--inline"
                      >
                        gitsync.selectBranchTitle
                      </span>
                    </label>
                    <div
                      class="bp3-form-group branchSelector"
                      data-id="branch-2"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-popover-wrapper Select--main bp3-fill"
                        >
                          <div
                            class="bp3-popover-target"
                          >
                            <div
                              class="bp3-input-group"
                            >
                              <input
                                autocomplete="off"
                                class="bp3-input"
                                name="branch"
                                placeholder="- Select -"
                                style="padding-right: 0px;"
                                type="text"
                                value=""
                              />
                              <span
                                class="bp3-input-action"
                              >
                                <span
                                  class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                                  icon="chevron-down"
                                >
                                  <svg
                                    data-icon="chevron-down"
                                    height="14"
                                    viewBox="0 0 16 16"
                                    width="14"
                                  >
                                    <desc>
                                      chevron-down
                                    </desc>
                                    <path
                                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                      fill-rule="evenodd"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-baseline StyledProps--flex-justifyContent-flex-start StyledProps--padding-top-small"
                  >
                    <div
                      class="bp3-form-group"
                      data-id="createPullRequest-3"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <label
                          class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main StyledProps--inline"
                        >
                          <input
                            name="createPullRequest"
                            type="checkbox"
                          />
                          <span
                            class="bp3-control-indicator"
                          />
                          <span
                            class="Tooltip--acenter FormikForm--checkBoxDocTooltipLabel"
                            data-tooltip-id="fullSyncConfigForm_createPullRequest"
                          >
                            common.git.startPRLabel
                          </span>
                        </label>
                      </div>
                    </div>
                    <div
                      class="bp3-form-group bp3-disabled branchSelector"
                      data-id="targetBranch-4"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-popover-wrapper Select--main bp3-fill"
                        >
                          <div
                            class="bp3-popover-target"
                          >
                            <div
                              class="bp3-input-group bp3-disabled"
                            >
                              <input
                                autocomplete="off"
                                class="bp3-input"
                                disabled=""
                                name="targetBranch"
                                placeholder="- Select -"
                                style="padding-right: 0px;"
                                type="text"
                                value=""
                              />
                              <span
                                class="bp3-input-action"
                              >
                                <span
                                  class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                                  icon="chevron-down"
                                >
                                  <svg
                                    data-icon="chevron-down"
                                    height="14"
                                    viewBox="0 0 16 16"
                                    width="14"
                                  >
                                    <desc>
                                      chevron-down
                                    </desc>
                                    <path
                                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                      fill-rule="evenodd"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="StyledProps--font StyledProps--main branchSection StyledProps--padding-bottom-small"
                >
                  <label
                    class="bp3-control bp3-radio bp3-large StyledProps--font Radio--radio StyledProps--main"
                  >
                    <input
                      data-test="newBranchRadioBtn"
                      type="radio"
                    />
                    <span
                      class="bp3-control-indicator"
                    />
                    <span
                      class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey700"
                      data-icon="git-new-branch"
                    >
                      <svg
                        height="16"
                        viewBox="0 0 13 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.406.003a2.399 2.399 0 00-.8 4.658v.94a.8.8 0 01-.8.8H4.001a2.413 2.413 0 00-.8.14V4.66A2.399 2.399 0 002.402 0a2.399 2.399 0 00-.8 4.661v6.678a2.399 2.399 0 00.8 4.661 2.399 2.399 0 00.8-4.661v-2.54a.8.8 0 01.8-.799h4.803a2.4 2.4 0 002.402-2.4v-.939a2.399 2.399 0 00-.8-4.658z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                    <span
                      class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--margin-left-small"
                    >
                      gitsync.createBranchTitle
                    </span>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-medium"
            >
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  save
                </span>
              </button>
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-left-medium Button--with-current-color"
                type="button"
              >
                <span
                  class="bp3-button-text"
                >
                  cancel
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
